praw/__init__.py

```diff
@@ -1689,16 +1691,19 @@ def upload_image(self, subreddit, image_path, name=None, header=False):
             size = os.path.getsize(image.name)
             if size < MIN_IMAGE_SIZE:
                 raise errors.ClientException('`image` is not a valid image')
-            elif size > MAX_IMAGE_SIZE:
+            if size > MAX_IMAGE_SIZE:
                 raise errors.ClientException('`image` is too big. Max: {0} '
                                              'bytes'.format(MAX_IMAGE_SIZE))
-            first_bytes = image.read(MIN_IMAGE_SIZE)
+            first_bytes = image.read(MIN_PNG_SIZE)
```

Comment:
read can instead use MIN_PNG_SIZE as it's sufficiently large. It could also be `len(PNG_HEADER)` as that's the largest that is compared.